simulator/src/VerilogClasses.js

```
@@ -66,16 +66,29 @@ import { newCircuit, switchCircuit, changeCircuitName} from './circuit'
+        return bitsJSON.length;
+    }
+    if (bitsJSON && typeof bitsJSON === "object") {
+        var widths = [];
+        for (var key in bitsJSON) {
+            var val = bitsJSON[key];
+            if (typeof val === "number") {
+                widths.push(val);
+            } else if (Array.isArray(val)) {
+                widths.push(val.length);
+            }
         }
```

ESLint

[error] for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array.

(no-restricted-syntax)